src/librustc_span/lib.rs

```diff
@@ -14,6 +14,10 @@
 #![feature(optin_builtin_traits)]
 #![feature(specialization)]
 
+// allow wasm target for rustc-ap-rustc_span
```

Comment:
This is not about WASM specifically, so maybe change this to

```suggestion
// FIXME(#56935): Work around ICEs during cross-compilation.
```